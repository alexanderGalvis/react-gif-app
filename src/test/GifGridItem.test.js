import { shallow } from 'enzyme/build';
import GifGridItem from '../components/GifGridItem';
import React from 'react';

describe('Pruebas GifGridItem.js', function () {
  const title = 'Un t√≠tulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Carga el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe tener un parrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('debe tener un url y alt', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('debe tener un class', () => {
    const div = wrapper.find('div');
    console.log(div.props);
    expect(div.prop('className')).toBe(
      'card animate__animated animate__fadeInDown animate__delay-1s'
    );
    expect(div.prop('className').includes('animate__fadeInDown')).toBe(true);
  });
});
