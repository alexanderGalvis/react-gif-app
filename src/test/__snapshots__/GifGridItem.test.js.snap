// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGridItem.js Carga el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown animate__delay-1s"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
